src/cryptonote_protocol/block_queue.cpp

```diff
@@ -256,8 +267,10 @@ void block_queue::set_span_hashes(uint64_t start_height, const boost::uuids::uui
     if (i->start_block_height == start_height && i->connection_id == connection_id)
     {
       span s = *i;
-      blocks.erase(i);
+      erase_block(i);
       s.hashes = std::move(hashes);
+      for (const crypto::hash &h: s.hashes)
+        requested_hashes.insert(h);
```

Comment:
IIUC, this inserts hash references after they are invalidated in `erase_block`.

Reply:
Nice catch, but hashes is not a reference (parameter to the function).